### ISSUE TITLE
step_feedback variables not in packet (1.1 version)

### ISSUE THREAD

**Issue by osrf-migration** [osrf/drcsim#263]
**Original report ([archived issue](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/issues/263)) by Chris Atkeson (Bitbucket: [catkeson](https://bitbucket.org/%7B3bdaa162-b747-4fa9-9d6e-a2d5c92933c6%7D/)).**

----------------------------------------
In AtlasSimInterface_1.1.0, it appears that the step_feedback section is not getting loaded.

When
~~~
current_behavior: 5, desired_behavior: 5:

step_feedback: 
  status_flags: 1
  t_step_rem: 0.0
  current_step_index: 0
  next_step_index_needed: 0
  desired_step_saturated: 
    step_index: 0
    foot_index: 0
    duration: 0.0
    pose: 
      position: 
        x: 0.0
        y: 0.0
        z: 0.0
      orientation: 
        x: 0.0
        y: 0.0
        z: 0.0
        w: 0.0
    swing_height: 0.0
~~~

~~~
rostopic echo /atlas/atlas_sim_interface_state
~~~
says
~~~
header: 
  seq: 96356
  stamp: 
    secs: 96
    nsecs: 422000000
  frame_id: ''
error_code: 0
current_behavior: 5
desired_behavior: 5
f_out: [-1.9673831462860107, -5.4964680671691895, 0.7754316329956055, 0.0, 1.6178072690963745, 18.054588317871094, 22.224016189575195, -29.231199264526367, 20.63804817199707, -1.0597435235977173, -0.01425538957118988, -2.0351486206054688, -52.04777145385742, -36.75736999511719, 17.120820999145508, -1.445822834968567, 3.6067428588867188, 11.446175575256348, -2.334134340286255, 2.8639931678771973, -0.19184903800487518, 1.391825795173645, 6.3044328689575195, -11.437204360961914, -1.7458786964416504, -0.8134133219718933, -0.12571981549263, -0.4682886302471161]
pos_est: 
  position: 
    x: 17.434129715
    y: 8.02688407898
    z: 0.903372228146
  velocity: 
    x: -0.0249147396535
    y: 0.019729135558
    z: 0.0227577965707
foot_pos_est: 
  - 
    position: 
      x: 17.5679645538
      y: 8.14220142365
      z: -0.000168442726135
    orientation: 
      x: -0.00404940871522
      y: 0.00473898323253
      z: 0.00909521430731
      w: 0.999939203262
  - 
    position: 
      x: 17.358587265
      y: 7.89121007919
      z: -0.000173032283783
    orientation: 
      x: -0.00404940871522
      y: 0.00473898323253
      z: 0.00909521430731
      w: 0.999939203262
behavior_feedback: 
  status_flags: 0
  trans_from_behavior_index: -1
  trans_to_behavior_index: -1
step_feedback: 
  status_flags: 1
  t_step_rem: 0.0
  current_step_index: 0
  next_step_index_needed: 0
  desired_step_saturated: 
    step_index: 0
    foot_index: 0
    duration: 0.0
    pose: 
      position: 
        x: 0.0
        y: 0.0
        z: 0.0
      orientation: 
        x: 0.0
        y: 0.0
        z: 0.0
        w: 0.0
    swing_height: 0.0
stand_feedback: 
  status_flags: 0
walk_feedback: 
  t_step_rem: 0.0
  current_step_index: 0
  next_step_index_needed: 1
  status_flags: 0
  step_queue_saturated: 
    - 
      step_index: 0
      foot_index: 4294967295
      duration: -1.0
      pose: 
        position: 
          x: 0.0
          y: 0.0
          z: 0.0
        orientation: 
          x: 0.0
          y: 0.0
          z: 0.0
          w: 1.0
      swing_height: -1.0
    - 
      step_index: 0
      foot_index: 4294967295
      duration: -1.0
      pose: 
        position: 
          x: 0.0
          y: 0.0
          z: 0.0
        orientation: 
          x: 0.0
          y: 0.0
          z: 0.0
          w: 1.0
      swing_height: -1.0
    - 
      step_index: 0
      foot_index: 4294967295
      duration: -1.0
      pose: 
        position: 
          x: 0.0
          y: 0.0
          z: 0.0
        orientation: 
          x: 0.0
          y: 0.0
          z: 0.0
          w: 1.0
      swing_height: -1.0
    - 
      step_index: 0
      foot_index: 4294967295
      duration: -1.0
      pose: 
        position: 
          x: 0.0
          y: 0.0
          z: 0.0
        orientation: 
          x: 0.0
          y: 0.0
          z: 0.0
          w: 1.0
      swing_height: -1.0
manipulate_feedback: 
  status_flags: 0
  clamped: 
    pelvis_height: 0.800000011921
    pelvis_yaw: 0.0
    pelvis_lat: 0.0
k_effort: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
~~~

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------

* Edited issue description

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------
it's missing, pull request coming.

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------

* changed state from "new" to "resolved"

fix issue #263, add step feedback, also update foot orientation quaternion based on normal vector.

→ \<\<cset 59fee803690eff7d68908e1731d58135ce6b706f\>\>

**Comment by osrf-migration** [comment]
**Original comment by John Hsu (Bitbucket: [hsu](https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/), GitHub: [hsu](https://github.com/hsu)).**

----------------------------------------
[pull request #275](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/pull-requests/275)
